File: 2020/2/solution2.py
with open("input1.txt") as input_file:
    valid_passwords = 0

    for line in input_file:
        policy, password = line.split(":")
        password = password.strip()
        indexes, char = policy.split()
        id1, id2 = indexes.split("-")
        two_chars = password[int(id1)-1] + password[int(id2)-1]
        if two_chars.count(char) == 1:
            valid_passwords += 1
    
    print(f"Valid passwords: {valid_passwords}")

File: 2020/2/solution2.go
package main

import (
    "fmt"
    "regexp"
    "io/ioutil"
    "strings"
    "strconv"
)


func main() {
    content, _ := ioutil.ReadFile("input1.txt")
    lines := strings.Split(string(content), "\n")
    regex := *regexp.MustCompile(`^(\d+)-(\d+) (\w): (\w+)$`)
    var valid_passwords int

    for _, line := range(lines) {
        if len(line) == 0 {
            continue
        }

        res := regex.FindAllStringSubmatch(line, -1)
        id1, _ := strconv.Atoi(res[0][1])
        id2, _ := strconv.Atoi(res[0][2])
        char := res[0][3]
        passwd := res[0][4]
        substr := string(passwd[id1-1]) + string(passwd[id2-1])

        count := strings.Count(substr, char)

        if count == 1 {
            valid_passwords += 1
        }
    }

    fmt.Println(valid_passwords)
}
